src/test/java/com/conveyal/r5/point_to_point/PointToPointRouterServerTest.java

```diff
@@ -0,0 +1,429 @@
+        ReadContext ctx = JsonPath.parse(response);
+
+        // get all polylines
+        JSONArray polylines = ctx.read("$.data.features");
+
+        // expect at least 1 polyline, likely many more
+        int numPolylines = polylines.size();
+        assertThat(numPolylines, greaterThan(0));
+
+
+        // expect an exact position of the first and last polylines
+        LinkedHashMap firstPolyline = (LinkedHashMap) polylines.get(0);
+        JSONArray firstPolylineCoordinates = (JSONArray)((LinkedHashMap)firstPolyline.get("geometry")).get("coordinates");
+        JSONArray firstPolylineFirstCoordinate = (JSONArray)firstPolylineCoordinates.get(0);
+        assertThat(firstPolylineFirstCoordinate.get(0), equalTo(-87.4099729));
+        assertThat(firstPolylineFirstCoordinate.get(1), equalTo(39.4654342));
+        LinkedHashMap lastPolyline = (LinkedHashMap) polylines.get(numPolylines - 1);
+        JSONArray lastPolylineCoordinates = (JSONArray)((LinkedHashMap)lastPolyline.get("geometry")).get("coordinates");
+        JSONArray lastPolylineLastCoordinate = (JSONArray)lastPolylineCoordinates.get(lastPolylineCoordinates.size() - 1);
+        assertThat(lastPolylineLastCoordinate.get(0), equalTo(-87.3837375));
+        assertThat(lastPolylineLastCoordinate.get(1), equalTo(39.4858394));
+
+        // expect each polyline to have a distance greater than the last
+        int curDistance = 0;
+        for (int i = 0; i < numPolylines; i++) {
+            int curPolylineDistance = (int)((LinkedHashMap)((LinkedHashMap)polylines.get(i)).get("properties")).get("distance");
+            assertThat(curPolylineDistance, greaterThanOrEqualTo(curDistance));
+            curDistance = curPolylineDistance;
+        }
+    }
+
+    /**
+     * Assert that a trip plan with walking can be dones
+     */
+    @Test
+    public void canMakeOTPGraphQLQuery() {
+        given()
+            .port(8080)
+            .body(makeGraphQLQuery())
+            .post("/otp/routers/default/index/graphql")
+        .then()
+            .body(matchesJsonSchemaInClasspath("com/conveyal/r5/point_to_point/otp-graphql-response.json"));
+    }
+
+    /**
+     * Make a GraphQL query to use in the body of a post request to /otp/routers/default/index/graphql
+     */
+    private String makeGraphQLQuery() {
+        String accessModes = "WALK,BICYCLE,BICYCLE_RENT,CAR_PARK";
+        String bikeSpeed = "8";
+        String bikeTrafficStress = "4";
+        String directModes = "CAR,WALK,BICYCLE,BICYCLE_RENT";
+        String egressModes = "WALK";
+        String fromLat = "39.465659";
+        String fromLon = "-87.410839";
+        String fromTime = "2017-09-18T12:00:00.000Z";
+        String toLat = "39.4767";
+        String toLon = "-87.4052";
+        String toTime = "2017-09-18T14:00:00.000Z";
+        String transitModes = "BUS,RAIL,SUBWAY,TRAM";
+        String walkSpeed = "3";
+
+        return "{" +
```

Comment:
Should this be stored as a resource file rather than a string literal? I find things like this less readable/maintainable when inlined into the code.

Reply:
Fixed in https://github.com/conveyal/r5/pull/345/commits/908d331ec291f2802cd484407d542022f5cd304d